Xamarin.PropertyEditing/Reflection/ReflectionEnumPropertyInfo.cs

```diff
@@ -149,6 +149,13 @@ private static Task<Delegate> GetOrAddHasFlagMethod<T> ()
 					generator.Emit (OpCodes.Ldarg_1);
 					generator.Emit (OpCodes.And); // arg0 & arg1
 					generator.Emit (OpCodes.Ldc_I4_0);
+					if (typeof (T) == typeof (uint))
+						generator.Emit (OpCodes.Conv_U4);
+					else if (typeof (T) == typeof (long))
+						generator.Emit (OpCodes.Conv_I8);
+					else if (typeof (T) == typeof (ulong))
+						generator.Emit (OpCodes.Conv_U8);
```

Comment:
`The approved types for an enum are byte, sbyte, short, ushort, int, uint, long, or ulong.`